### PR TITLE
JBIDE-22325 - Move job to disable all widgets into foundation, and al…

### PR DESCRIPTION
**Pull request by robstryker**
…low astools to use it

### CODE REVIEW

**Comment by fbricon** [comment]
Where are the unit tests?

**Comment by mickaelistria** [comment]
A missing "final" modifier on the control parameter in WidgetUtility makes the build fail: http://jenkins.mw.lab.eng.bos.redhat.com/hudson/job/jbosstools-base_master/940/console
Please fix it ASAP.

File: foundation/plugins/org.jboss.tools.foundation.ui/src/org/jboss/tools/foundation/ui/jobs/DisableAllWidgetsJob.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.swt.graphics.Cursor;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.ui.progress.UIJob;
+import org.jboss.tools.foundation.ui.widget.WidgetUtility;
+
+public class DisableAllWidgetsJob extends UIJob {
```
Comment (maxandersen) [inline foundation/plugins/org.jboss.tools.foundation.ui/src/org/jboss/tools/foundation/ui/jobs/DisableAllWidgetsJob.java]: no javadoc. agreement was all public classes in foundation should have good docs and tests.

File: foundation/plugins/org.jboss.tools.foundation.ui/src/org/jboss/tools/foundation/ui/widget/IWidgetVisitor.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  * and is available at http://www.eclipse.org/legal/epl-v10.html
+  *
+  * Contributor:
+  *     Red Hat, Inc. - initial API and implementation
+  ******************************************************************************/
+package org.jboss.tools.foundation.ui.widget;
+
+import org.eclipse.swt.widgets.Control;
+
+public interface IWidgetVisitor {
```
Comment (maxandersen) [inline foundation/plugins/org.jboss.tools.foundation.ui/src/org/jboss/tools/foundation/ui/widget/IWidgetVisitor.java]: no javadocs

File: foundation/plugins/org.jboss.tools.foundation.ui/src/org/jboss/tools/foundation/ui/widget/WidgetUtility.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  *
+  * Contributor:
+  *     Red Hat, Inc. - initial API and implementation
+  ******************************************************************************/
+package org.jboss.tools.foundation.ui.widget;
+
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Control;
+
+public class WidgetUtility {
```
Comment (maxandersen) [inline foundation/plugins/org.jboss.tools.foundation.ui/src/org/jboss/tools/foundation/ui/widget/WidgetUtility.java]: no javadocs

**Pull request merged**